students/jason_jenkins/lesson03/mailroom.py

```diff
@@ -9,57 +9,151 @@
+            print()
         else:
             break
 
+    if(response != "exit"):
+        for donor in donor_list:
+            if donor[0] == response:
+                donate(donor)
+                break
+        else:
+            new_donor = [response]
+            donate(new_donor)
+            donor_list.append(new_donor)
+
+
+def thank_you_email(donor):
+    """
+    Thank donor for donation
+    """
+
+    print(f"Thank you {donor[0]} for your donation.")
+
+
+def print_donor_list():
+    """
+    Print out the donor list
+    """
+
+    global donor_list
+
+    print("List of donors")
+    print("--------------")
+
+    for donor in donor_list:
+        print(donor[0])
+
+
+def donate(donor):
+    """
+    Prompt donor to donate
+    """
+
+    response = float(input('Input amount to donate or "0" to exit: '))
+    if(response != 0):
+        donor.append(response)
+        thank_you_email(donor)
+
 
-def donate():
-    print("add donation")
+def sort_by_total(tmp_donor_list):
+    """
+    Used to sort donor list by total donations
+    """
+
+    return sum(tmp_donor_list[1:])
 
 
 def create_report():
-    print("create_report")
+    """
+    Create a table like view of donors
+    Includes donor name, donation total, total gifts, average gift amount
 
+    Sorted by donation total
+    """
+
+    global donor_list
+
+    donor_list.sort(key=sort_by_total, reverse=True)
+
+    print(f"{'Donor Name':30}|{'Total Given':^16}|", end='')
+    print(f"{'Num Gifts':^14}|{'Average Gift':^16}")
+    print(f"{'-'*79}")
+
+    for donor in donor_list:
+        donor_name = donor[0]
+        donor_total = sum(donor[1:])
+        donar_count = len(donor) - 1
+        donor_ave = 0
+
+        if donar_count != 0:
+            donor_ave = donor_total / donar_count
+
+        donor_output = f"{donor_name:30}"
+        donor_output += f" ${donor_total:15.2f}"
+        donor_output += f"{donar_count:15}"
+        donor_output += f" ${donor_ave:15.2f}"
+        print(donor_output)
+
+
+def quit_program():
+    """
+    Method used to quit the program
+    """
 
-def startup_prompt():
     global run_program
+    run_program = False
+
+
+def startup_prompt():
+    """
+    Prombt user for action they what to take
+    """
 
     print()
     print("Do you want to:")
-    print("\t1. Send a thank you")
-    print("\t2. Create a report")
-    print("\t3. Quit")
+    print('   "Send a thank you"')
+    print('   "Create a Report"')
+    print('   "quit"')
 
-    response = input("Input number option you wish to do: ")
+    response = input("Input option you wish to do: ")
 
-    # Add Error checking
-    response = int(response)
-    if(response == 1):
+    if(response.lower() == "send a thank you"):
         send_thanks()
-    elif(response == 2):
+    elif(response.lower() == "create a report"):
         create_report()
-    elif(response == 3):
-        run_program = False
+    elif(response.lower() == "quit"):
+        quit_program()
```

Comment:
instead of calling lower() every time you can just do `response = input("Input option you wish to do: ").lower()`